pkg/networkmanager/firewall.jsx

```diff
@@ -471,39 +484,19 @@ class AddServicesModal extends React.Component {
         else
             services = this.state.services;
 
-        // hide services which have been enabled in all zones
-        if (services) {
-            services = services.filter(s => {
-                let allZonesContainService = true;
-                for (const zone of firewall.activeZones)
-                    allZonesContainService &= firewall.zones[zone].services.indexOf(s.id) !== -1;
-                return !allZonesContainService;
-            });
-        }
+        // hide services which have been enabled in the zone
+        if (services)
+            services = services.filter(s => firewall.zones[this.props.zoneId].services.indexOf(s.id) === -1);
 
-        var addText = this.state.custom ? _("Add Ports") : _("Add Services");
-        var zonesText = this.state.custom ? _("Add ports to the following zones:") : _("Add services to following zones:");
+        const addText = this.state.custom ? _("Add Ports") : _("Add Services");
+        const titleText = cockpit.format(_("$0 to $1 zone"), addText, this.props.zoneName);
```

Comment:
This may get "interesting" translations in some languages.
It would be better to have two strings like 'Add Ports to $0 zone' and 'Add Services to $0 zone'. Also should be 'Zone' capitalized or Services/Ports start with small letter?

Reply:
Yea that make sense, didn't realize this would come out totally garbled in some languages.